### PR TITLE
Hw intro Spring

### PR DESCRIPTION
**Pull request by Artsapollo**

### CODE REVIEW

File: src/main/java/introSpring/hw/garageConfig/GarageAppConfig.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    @Bean("Nsx")
+    public Nsx returnNsx(){
+        return new Nsx();
+    }
+
+    @Bean("Supra")
+    public Supra returnSupra(){
+        return new Supra();
+    }
+}
```
Comment (Yanetta) [inline src/main/java/introSpring/hw/garageConfig/GarageAppConfig.java]: As I understood HW , implementation of beans should differ from components implementation.

File: src/main/java/introSpring/hw/Evo.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.stereotype.Component;
+
+@Component("Evo")
+public class Evo extends Garage {
+    @Override
+    public String getName() {
+        return "Mistubishi Lancer Evolution VIII";
+    }
+
+    @Override
```
Comment (Yanetta) [inline src/main/java/introSpring/hw/Evo.java]: What is the reason to override this method, if you use parent implementation?

File: src/main/java/introSpring/hw/Garage.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package introSpring.hw;
+
+public abstract class Garage implements Car{
+    @Override
+    public String getName() {
+        return this.getClass().getSimpleName();
+    }
+
+    @Override
```
Comment (Yanetta) [inline src/main/java/introSpring/hw/Garage.java]: Does this method calls itself?

Comment (Yanetta) [reply src/main/java/introSpring/hw/Garage.java]: looks like recursion

Comment (Artsapollo) [reply src/main/java/introSpring/hw/Garage.java]: You pointed me on problem I can't find, thank you

**Pull request merged**